src/hooks/useShowPhase.ts

```diff
@@ -0,0 +1,18 @@
+import { CogsClientMessage, CogsConnection, ShowPhase } from '@clockworkdog/cogs-client';
+import { useCallback, useState } from 'react';
+import useCogsMessage from './useCogsMessage';
+
+export default function useShowPhase(connection: CogsConnection): ShowPhase {
+  const [status, setStatus] = useState<ShowPhase>(ShowPhase.Setup);
```

Comment:
Similar to the thing with hints, we should probably make the connection cache the current status so it can be used as an initial default here? Then you can conditionally render components with this hook. From the COGS side I'm thinking we should then also send a `show_phase` message on first connection so the client can be guaranteed to be incorrect state regardless of when it connects.

Reply:
> Similar to the thing with hints, we should probably make the connection cache the current status so it can be used as an initial default here? Then you can conditionally render components with this hook.

Yeah for sure! Done.